msdsales/parse.go

```diff
@@ -0,0 +1,40 @@
+package msdsales
+
+import (
+	"github.com/amp-labs/connectors/common"
+	"github.com/spyzhov/ajson"
+)
+
+func getTotalSize(node *ajson.Node) (int64, error) {
+	return common.JSONManager.ArrSize(node, "value")
+}
+
+func getRecords(node *ajson.Node) ([]map[string]any, error) {
+	arr, err := common.JSONManager.GetArr(node, "value")
+	if err != nil {
+		return nil, err
+	}
+
+	return common.JSONManager.ArrToMap(arr)
+}
+
+func getNextRecordsURL(node *ajson.Node) (string, error) {
+	return common.JSONManager.GetString(node, "@odata.nextLink")
+}
+
+// FIXME we must differentiate between GET and LIST (it is LIST now)
+// sometimes the fields that user requests are either
+// * singular record
+// * list of records
+// * hybrid, list of records with extra fields describing list.
+func getMarshaledData(records []map[string]interface{}, fields []string) ([]common.ReadResultRow, error) {
```

Comment:
Can we marshal singular record in case of 
`GET domain/cats/<catID>` vs `GET domain/cats`
Also LIST of cats could return additional parameters stored outside of array, How those fields are returned?